controllers/operands/wasp.go

```diff
@@ -0,0 +1,449 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	// For simplicity matters, re-use the virt-handler SA. That will spare
+	// the HCO controller from deploying custom privileged SA with SCC.
+	waspServiceAccount = "kubevirt-handler"
+	// Set dry run mode for e2e testing purposes
+	waspDryRunAnnotation = "wasp.hyperconverged.io/dry-run"
+)
+
+func newWaspHandler(Client client.Client, Scheme *runtime.Scheme) Operand {
+	return &conditionalHandler{
+		operand: &genericOperand{
+			Client: Client,
+			Scheme: Scheme,
+			crType: "DaemonSet",
+			hooks:  &waspHooks{},
+		},
+		getCRWithName: func(hc *hcov1beta1.HyperConverged) client.Object {
+			return NewWaspWithNameOnly(hc)
+		},
+		shouldDeploy: func(hc *hcov1beta1.HyperConverged) bool {
+			// convert from anon to private
+			return hc.Spec.FeatureGates.EnableHigherDensityWithSwap != nil &&
+				*hc.Spec.FeatureGates.EnableHigherDensityWithSwap
+		},
+	}
+}
+
+type waspHooks struct {
+	cache *appsv1.DaemonSet
+}
+
+func (h *waspHooks) getFullCr(hc *hcov1beta1.HyperConverged) (client.Object, error) {
+	if h.cache == nil {
+		h.cache = NewWasp(hc)
+	}
+
+	return h.cache, nil
+}
+
+func (*waspHooks) getEmptyCr() client.Object { return &appsv1.DaemonSet{} }
+
+func (*waspHooks) updateCr(
+	req *common.HcoRequest,
+	Client client.Client,
+	exists runtime.Object,
+	required runtime.Object) (bool, bool, error) {
+	daemonset, ok1 := required.(*appsv1.DaemonSet)
+	found, ok2 := exists.(*appsv1.DaemonSet)
+
+	if !ok1 || !ok2 {
+		return false, false, errors.New("can't convert to Daemonset")
+	}
+	if !hasCorrectDaemonSetFields(found, daemonset) {
+		if req.HCOTriggered {
+			req.Logger.Info("Updating existing wasp daemonset to new opinionated values", "name", daemonset.Name)
+		} else {
+			req.Logger.Info("Reconciling an externally updated wasp daemonset to its opinionated values", "name", daemonset.Name)
+		}
+		if shouldRecreateDaemonset(found, daemonset) {
+			err := Client.Delete(req.Ctx, found, &client.DeleteOptions{})
+			if err != nil {
+				return false, false, err
+			}
+			err = Client.Create(req.Ctx, daemonset, &client.CreateOptions{})
+			if err != nil {
+				return false, false, err
+			}
+			return true, !req.HCOTriggered, nil
+		}
+		hcoutil.MergeLabels(&daemonset.ObjectMeta, &found.ObjectMeta)
+		daemonset.Spec.DeepCopyInto(&found.Spec)
+		err := Client.Update(req.Ctx, found)
+		if err != nil {
+			return false, false, err
+		}
+		return true, !req.HCOTriggered, nil
+	}
+
+	return false, false, nil
+}
+
+func (*waspHooks) justBeforeComplete(_ *common.HcoRequest) { /* no implementation */ }
+
+func (*waspHooks) getConditions(cr runtime.Object) []metav1.Condition {
```

Comment:
This is a partial implementation of the `hcoOperandHooks` interface. This interface includes two methods, and this PR implements only one of them. That why the linter says this method is unused, because it never called for types that are not fully implementing the interface. 

But actually, we don't need this method at all. this method should collect the conditions from KubeVirt CRs, that are publishing common list of conditions. The DaemonSet does not publish the KV common conditions, obviously, and the is no point to implement the interface.

However, maybe it will be a good idea to add a new condition to HCO to reflect the status of the swap daemonset? It won't be simple though, so we can add it in a following PR. @enp0s3 , @orenc1 , @tiraboschi  - WDYT?